pkg/dynamicrp/backend/service.go

```diff
@@ -18,56 +18,67 @@ package backend
 
 import (
 	"context"
-	"fmt"
 
 	"github.com/radius-project/radius/pkg/armrpc/asyncoperation/worker"
-	"github.com/radius-project/radius/pkg/armrpc/hostoptions"
 	"github.com/radius-project/radius/pkg/dynamicrp"
+	"github.com/radius-project/radius/pkg/recipes/controllerconfig"
 )
 
 // Service runs the backend for the dynamic-rp.
 type Service struct {
 	worker.Service
 	options *dynamicrp.Options
+	recipes *controllerconfig.RecipeControllerConfig
 }
 
 // NewService creates a new service to run the dynamic-rp backend.
 func NewService(options *dynamicrp.Options) *Service {
 	return &Service{
 		options: options,
 		Service: worker.Service{
-			ProviderName: "dynamic-rp",
-			Options: hostoptions.HostOptions{
-				Config: &hostoptions.ProviderConfig{
-					Env:              options.Config.Environment,
-					DatabaseProvider: options.Config.Database,
-					SecretProvider:   options.Config.Secrets,
-					QueueProvider:    options.Config.Queue,
-				},
-			},
+			// Will be initialized later
 		},
+		recipes: options.Recipes,
 	}
 }
 
 // Name returns the name of the service used for logging.
 func (w *Service) Name() string {
-	return fmt.Sprintf("%s async worker", w.Service.ProviderName)
+	return "dynamic-rp async worker"
 }
 
 // Run runs the service.
 func (w *Service) Run(ctx context.Context) error {
-	err := w.Init(ctx)
+	if w.options.Config.Worker.MaxOperationConcurrency != nil {
```

Comment:
This was simplified based on changes we made in the database PR. I'm still not a big fan of the layering here, but we need to move on.